File: 5-kyu/rot13-1/rot13-1.js
function rot13(message) {
  return message.split('').map(char => {
    const code = char.charCodeAt(0);
​
    if (code >= 65 && code <= 90) {
      // Uppercase A-Z
      return String.fromCharCode(((code - 65 + 13) % 26) + 65);
    }
    if (code >= 97 && code <= 122) {
      // Lowercase a-z
      return String.fromCharCode(((code - 97 + 13) % 26) + 97);
    }
    return char; // Non-letter
  }).join('');
}
​
//initial; bad performance
// function rot13(message){
//   const alphabet = 'abcdefghijklmnopqrstuvwxyz';
//   //your code here
//   return message.split('').map( char => {
//     if(/[^A-Za-z]/.test(char)) return char;
//     const isUpper = char >= 'A' && char <= 'Z';
//     let index = alphabet.indexOf(char.toLowerCase());
//     let finalChar = alphabet[(index + 13) % 26];
//     return isUpper ? finalChar.toUpperCase() : finalChar;
//   }).join('');
// }

File: 5-kyu/rot13-1/rot13-1.py
def rot13(text):
    result = []
​
    for char in text:
        if 'a' <= char <= 'z':
            # Shift lowercase letters
            shifted = chr((ord(char) - ord('a') + 13) % 26 + ord('a'))
            result.append(shifted)
        elif 'A' <= char <= 'Z':
            # Shift uppercase letters
            shifted = chr((ord(char) - ord('A') + 13) % 26 + ord('A'))
            result.append(shifted)
        else:
            # Leave non-letters unchanged
            result.append(char)
​
    return ''.join(result)